pywatts/summaries/tsne_visualisation.py

```diff
@@ -0,0 +1,134 @@
+from typing import Dict
+
+import matplotlib.pyplot as plt
+import tikzplotlib
+import xarray as xr
+from pywatts.core.base_summary import BaseSummary
+from sklearn.manifold import TSNE
+import numpy as np
+
+from pywatts.core.summary_object import SummaryObjectList
+
+
+class TSNESummary(BaseSummary):
+    """
+    Summary that performs a TSNE to visualise the data. It is possible to specify mask by providing kwargs
+    to the transform method that end with _masked.
+    :param max_point: The maximum number of points per data that should be plotted
+    :type max_point: int
+    :param all_in_one_plot: Flag indicating if all input data should be visualised in the same plot. If not the column
+                            GT is visualised with all other in separate plots.
+    :type all_in_one_plot: Boolean
+    :param tsne_params: Params for the TSNE Visualisation. See sklearn.
```

Comment:
Params for the t-SNE visualisation (see sklearn at https://scikit-learn.org/stable/modules/generated/sklearn.manifold.TSNE.html).